### DISCUSSION TITLE
직원들의 로그인과 관련된 사항

### DISCUSSION THREAD

**Discussion by Topslug** [cau-swe/checkpoint-1#22] [General]
로그인 과정에 대해 다른 분들의 의견이 궁금해서 제 생각을 공유드립니다.

저는 '로그인 과정에서 직원이 반드시 개인 당 하나의 계정을 가져야 하는가?'에 대해서 조금 의문이 있었습니다.
어차피 매장 컴퓨터는 직원들이 1(컴퓨터)대 다(직원)로 이용하는 것이고 '직원 누가 접속했는가'가 중요한 것이 아니라 '직원이 접속했는가'가 중요한 것이라는 생각이 들었기 때문입니다. 
그래서 저는 '직원 계정을 하나로 통일해서 직원들이 공유해서 쓰는 걸로 해도 괜찮지 않을까?'라는 생각을 하게 되었습니다.

이런 의문이 들었던 이유는 약간 모순되는 상황이 발생할 수밖에 없다는 생각이 들었기 때문입니다.
관리자가 직원을 등록해서 사용할 건데 등록 시 계정 정보를 직원이 아닌 관리자가 입력해줘야 하는지, 그렇다면 직원 본인 확인 용 개인 정보인데, 직원이 아닌 다른 사람(관리자가)이 알게 되는 상황이 발생하게 됩니다.
그런데 관리자가 직원을 등록해서 사용하는 방식이라면 이런 상황은 피할 수 없는 것이고, 사장이 각각의 직원들에게 다른 계정 정보를 알려주게 되는 상황이라면 아예 직원들의 계정을 하나로 통일해서 사용해도 되지 않을까?라는 생각이 들게 되었습니다.
그렇게 한다면 사장과 직원들만 계정을 공유하게 되는 것이고, 사장이 직원 계정을 관리하는 것이 어색한 일이 아니게 되는 것 같습니다.
또한 직원들의 계정을 하나로 통일하게 된다면 본인 인증보다 직원 인증이라는 개념이 더 강화된다는 측면에서 관리자와 직원관계가 더 명확해진다는 생각이 듭니다.

그리고 이런 방식을 따른다면 어제 제기됐었던 고객과 직원이 같은 시스템에 접속했을 때 페이지 구분을 어떻게 하게 될 것 인가에 대한 문제도 해결되지 않을까 싶습니다.
고객과 직원을 완전하게 분리해서 관리할 수 있게 되서, 직원이 고객으로 가입해도 문제가 없고, 고객이었던 사람이 직원으로 등록돼도 문제가 없을 것 같습니다.
(그런데 저는 개인적으로 매장 컴퓨터에서 접속 시 바로 로그인 화면이 뜨는 것은 사용자 편의 측면에서 괜찮은 것 같긴 합니다.)

그래서 이 부분에 대해 다른 분들의 생각을 들어보고 싶습니다.

**Comment by Jummi10** [comment]
정리를 하자면 
1. 직원들의 정보를 관리자가 알아선 안 된다. 관리자가 직원을 등록하는 방식을 사용하면 안 된다.
2. 모든 직원들은 하나의 계정을 사용하여 주문 관리 페이지에 접속하게 한다.
3. 직원들이 주문 관리 페이지를 사용할 때에 로그인 화면이 바로 떠야한다.

총 이 세 가지 의견이 맞나요?


1. 먼저 궁금한 점이 있습니다.
> 관리자가 직원을 등록해서 사용할 건데 등록 시 계정 정보를 직원이 아닌 관리자가 입력해줘야 하는지, 그렇다면 직원 본인 확인 용 개인 정보인데, 직원이 아닌 다른 사람(관리자가)이 알게 되는 상황이 발생하게 됩니다.

이 부분이 모순적이라는 것인가요?? 그렇다면 앞 문장에 관해서는 이미 관리자가 직원 정보를 등록한다고 얘기가 되었었던 것 같습니다. 뒷 문장에 관해서도 현실세계에 빗대어 생각을 해본다면 관리자는 직원의 개인 정보(이름, 성별, 나이, 핸드폰 번호, 계좌 번호, 월급 등등이 있겠죠)를 가지고 있는 것이 당연하다고 생각이 듭니다. 직원 한 명을 고용한다(혹은 직원 한 명이 일을 그만둔다) = 직원 정보를 관리자가 등록해야 한다(관리자가 해당 직원의 정보를 삭제해야한다) 라고 보여져요. 어찌되었든 관리자가 직원들의 정보를 가지고 있어야한다는 점은 뺄 수 없는 부분인 것 같습니다.
관리자가 직원은 직원으로서 따로 관리하고, 고객으로서도 따로 가입을 할 수 있게 한다면 분리가 되지 않을까요. 피자 시스템이라는 전체 시스템 내에 한 사람이 직원으로서의 데이터와 고객으로서의 데이터를 따로 들고 있게 되는 것이죠.

2. `'직원 누가 접속했는가'가 중요한 것이 아니라 '직원이 접속했는가'가 중요한 것`라는 의견에 대해서는 저는 동의하는 바입니다. 하지만 
> 사장과 직원들만 계정을 공유하게 되는 것이고, 사장이 직원 계정을 관리하는 것이 어색한 일이 아니게 되는 것 같습니다.

이 부분에 대해서, 어떠한 이유로 이렇게 느끼셨는지 궁금합니다! 직원들이 여러 계정을 이용하며 사장님이 그 계정들을 관리하는 것은 어색하고, 하나의 계정을 여러 직원들이 공유해서 사용함으로써 관리자가 직원 계정을 관리하는 것은 어색하지 않은 것인가요??

**Reply by Topslug** [reply]
> 관리자가 직원을 등록해서 사용할 건데 등록 시 계정 정보를 직원이 아닌 관리자가 입력해줘야 하는지, 그렇다면 직원 본인 확인 용 개인 정보인데, 직원이 아닌 다른 사람(관리자가)이 알게 되는 상황이 발생하게 됩니다.

> 이 부분이 모순적이라는 것인가요??

제가 모순적이라고 생각했던 것은 관리자가 어차피 직원들의 계정을 알게 되고, 직원은 매장컴퓨터 하나를 이용하게 되는 상황인데, 관리자가 직원 개개인에게 각각 다른 ID와 비밀번호를 부여하게 된다면, 관리자 입장에서는 여러 개의 직원계정을 관리하기 어렵지 않을까 싶은 상황이었습니다. 직원을 추가할 때마다 새로운 계정 아이디와, 비밀번호를 생각해야하기 때문입니다. 또한 직원 입장에서도 굳이 개개인의 계정이 필요하지 않은 것이, 직원이 시스템에 접속하는 것은 주문 처리를 하기 위함인데, 직원 A가 접속하나, 직원 B가 접속하나 똑같기 때문입니다.(나중에 누가 주문 처리를 했는가! 라고 책임 추궁이 필요하다면 필요할 수도 있겠군요, 하지만 너무 복잡해질 것 같으니 생략하는 게 좋을 것 같습니다.)

> '직원 누가 접속했는가'가 중요한 것이 아니라 '직원이 접속했는가'가 중요한 것라는 의견에 대해서는 저는 동의하는 바입니다.

그래서 이러한 점에 비추어 볼 때, 직원 계정은 하나로 통일해도 되지 않을까라는 생각을 하게 된 것입니다.

> 직원들의 정보를 관리자가 알아선 안 된다. 관리자가 직원을 등록하는 방식을 사용하면 안 된다.

이 부분은 저의 표현 미숙으로 인해 잘못 이해하신 것 같습니다.
직원 계정의 통일을 하더라도 관리자는 여전히 직원등록, 정보수정, 삭제가 가능하며 직원계정관리(ID, 비밀번호)와 직원정보관리(이름, 전화번호, 월급)를 분리하겠다는 의도였습니다.

직원계정을 하나로 통일하는 것과 관리자가 직원들의 정보를 보고 관리하는 것은 서로 독립적이라고 생각합니다.
직원계정을 하나로 통일하는 것은 관리자가 직원 계정을 관리할 때의 편의성을 고려한 것입니다.
그래서 직원계정을 하나로 통일한다고 해도 관리자는 여전히 직원들의 정보(이름, 전화번호, 월급 등)을 보고 관리할 수 있습니다.(계정은 집에 들어갈 때 사용하는 키?라는 생각이 듭니다.)

제가 표현을 잘못한 것 같습니다. 죄송합니다.

> 그렇다면 직원 본인 확인 용 개인 정보인데, 직원이 아닌 다른 사람(관리자가)이 알게 되는 상황이 발생하게 됩니다.

이 부분에서 오해가 발생한 것 같은데
저는 여기서 본인 확인 용 개인 정보를 로그인 계정 정보(ID, 비밀번호)로만 한정시켜서 생각한 것입니다. 직원정보(이름, 전화번호, 월급 등)와는 관련이 없습니다.

다만, 제가 전에는 미처 생각하지 못했던, 관리자 계정 설정 부분과 , 직원 계정 설정 부분이 추가되어야 할 것 같습니다.

> 관리자가 직원은 직원으로서 따로 관리하고, 고객으로서도 따로 가입을 할 수 있게 한다면 분리가 되지 않을까요. 피자 시스템이라는 전체 시스템 내에 한 사람이 직원으로서의 데이터와 고객으로서의 데이터를 따로 들고 있게 되는 것이죠.

이 부분에 대해서 동의합니다. 그래서 따로 관리하는 것으로 생각하고 있었습니다.
지적해 주신 것처럼, 이미 직원 등록과 회원 가입은 분리를 시켜 놨었습니다.
다만 아직 고객 리스트를 확인하는 인터페이스는 추가하지 못한 상황입니다.
추후에 추가하도록 하겠습니다. 
(사실 이 부분에 대해서는 굳이 추가를 해야 하는가 의문을 가지고 있는 상황인데, 왜냐하면 고객 계정을 따로 관리하고, 고객이었던 사람이 직원이 된다고 해도 문제가 발생하지 않는 상황이라면, 고객 리스트를 확인해봤자 할 수 있는 것이 아무것도 없기 때문입니다. 회원을 받기는 하니 뭔가 관리자와 고객 간에 상호작용이 필요할 것 같긴 한데, Requirement에도 해당 사항이 없기도 하고, 이 부분은 관리자 쪽이 아닌 다른 팀원들의 의견이 중요할 것 같습니다.)

> 사장과 직원들만 계정을 공유하게 되는 것이고, 사장이 직원 계정을 관리하는 것이 어색한 일이 아니게 되는 것 같습니다.

> 이 부분에 대해서, 어떠한 이유로 이렇게 느끼셨는지 궁금합니다! 직원들이 여러 계정을 이용하며 사장님이 그 계정들을 관리하는 것은 어색하고, 하나의 계정을 여러 직원들이 공유해서 사용함으로써 관리자가 직원 계정을 관리하는 것은 어색하지 않은 것인가요??

이 부분도 저의 표현 미숙으로 인한 결과인 것 같습니다.
제가 어색하다고 한 것은 앞서 설명 드렸던, 사장님이 직원들에게 ID, 비밀번호를 알려줄 수 밖에 없는 상황이라면, 개개인의 ID, 비밀번호를 만드는 상황은 불편(= 어색)할 것 같다는 것이었습니다.

**Reply by Topslug** [reply]
생각해보니 Employee쪽의 Requirement에 직원들은 본인들의 이름, 전화번호 등을 시스템에서 확인할 수 있다와 같은 내용이 없으므로 직원들이 본인 계정을 가지고 있는 것의 의미가 조금 더 희석되는 것 같습니다.

**Reply by Topslug** [reply]
그리고 직원 계정을 하나로 통일해도 Employee쪽의 Use Case에는 아마 영향이 거의 없지 않을까요..? 고객과는 아무런 상관이 없을 것 같습니다.
